src/Functions/FunctionStrncmp.cpp

```diff
@@ -0,0 +1,445 @@
+#include <algorithm>
+#include <limits>
+#include <type_traits>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnLowCardinality.h>
+#include <Columns/ColumnSparse.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <Core/ColumnsWithTypeAndName.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/IDataType.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsComparison.h>
+#include <Functions/IFunction.h>
+#include <Interpreters/Context.h>
+#include <Common/memcmpSmall.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int ILLEGAL_COLUMN;
+}
+
+// usage:
+// strncmp(s, offset_s, t, offset_t, n)
+//  Compare s from offset_s to offset_s + n with t from offset_t to offset_t + n.
+class FunctionStrncmp : public IFunction
+{
+public:
+    static constexpr auto name = "strncmp";
+    static FunctionPtr create(const ContextPtr /*context*/) { return std::make_shared<FunctionStrncmp>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 5; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes &) const override { return std::make_shared<DataTypeInt8>(); }
+
+    DataTypePtr getReturnTypeForDefaultImplementationForDynamic() const override { return std::make_shared<DataTypeInt8>(); }
+
+    ColumnPtr
+    executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & /*result_type*/, size_t /*input_rows_count*/) const override
+    {
+        checkArguments(arguments);
+        auto compare_result_column = ColumnInt8::create();
+        compare_result_column->getData().resize(arguments[0].column->size());
+
+        auto first_str_col = arguments[0].column;
```

Comment:
Is it really necessary to remove LC and sparse? Are such columns not unpacked already by the function framework?